tests/mimc.rs

```diff
@@ -226,4 +231,37 @@ fn test_mimc() {
 
     println!("Average proving time: {:?} seconds", proving_avg);
     println!("Average verifying time: {:?} seconds", verifying_avg);
+
+    // batch verification
+    {
+        let pvk = prepare_batch_verifying_key(&params.vk);
+
+        let start = Instant::now();
+        let proofs: Vec<_> = proofs.iter().collect();
+        let valid = verify_proofs_batch(&pvk, &mut rand::rngs::OsRng, &proofs, &images).unwrap();
+        println!(
+            "Batch verification of {} proofs: {:?} seconds",
+            proofs.len(),
+            (start.elapsed().subsec_nanos() as f64) / 1_000_000_000f64,
+        );
+        assert!(valid, "failed batch verification");
+
+        // check that invalid proofs don't validate
+        let bad_proofs: Vec<Proof<Bls12>> = proofs
```

Comment:
an additional test where only one proof in batch is bad would feel more convincing

Reply:
e.g. maybe to proofs vector from previous test, and add one bad proof at a random index